tools/wasi-headers/src/c_header.rs

```diff
@@ -38,6 +38,7 @@ _Static_assert(_Alignof(int32_t) == 4, "non-wasi data layout");
 _Static_assert(_Alignof(uint32_t) == 4, "non-wasi data layout");
 _Static_assert(_Alignof(int64_t) == 8, "non-wasi data layout");
 _Static_assert(_Alignof(uint64_t) == 8, "non-wasi data layout");
+_Static_assert(_Alignof(void*) == 4, "non-wasi data layout");
```

Comment:
It's worth adding a comment somewhere around here mentioning that some of these asserts will need to be generalized when wasm64 happens.

Reply:
Agreed, all of the layout calculations will need to be generalized when that happens, but I'll add a comment so nobody misses it if they're consuming this file.